src/drv_psvr/psvr.c

```diff
@@ -64,30 +78,34 @@ static void handle_tracker_sensor_msg(psvr_priv* priv, unsigned char* buffer, in
 	psvr_sensor_packet* s = &priv->sensor;
 
 	uint32_t tick_delta = 500;
-	if(last_sample_tick > 0){ //startup correction
-		tick_delta = s->samples[0].tick - last_sample_tick;
-		if (tick_delta > 0xffffff)
-			tick_delta += 0x1000000;
-		if (tick_delta < 475 || tick_delta > 525){
-			LOGD("tick_delta = %u\n", tick_delta)
+
+	// Startup correction, ignore last_sample_tick if zero.
+	if (last_sample_tick > 0) {
+		tick_delta = calc_delta_and_handle_rollover(
+			s->samples[0].tick, last_sample_tick);
+
+		// The PSVR device can buffer sensor data from previous
+		// sessions which we can get at the start of new sessions.
+		// @todo Maybe just skip the first 10 sensor packets?
+		// @todo Maybe reset sensor fusion?
+		if (tick_delta < 475 || tick_delta > 525) {
+			LOGD("tick_delta = %u", tick_delta);
 			tick_delta = 500;
 		}
 	}
 
-	float dt = tick_delta * TICK_LEN;
 	vec3f mag = {{0.0f, 0.0f, 0.0f}};
 
-	for(int i = 0; i < 2; i++){
-		accel_from_psvr_vec(s->samples[0].accel, &priv->raw_accel);
-		gyro_from_psvr_vec(s->samples[0].gyro, &priv->raw_gyro);
+	for (int i = 0; i < 2; i++) {
+		float dt = tick_delta * TICK_LEN;
+		accel_from_psvr_vec(s->samples[1].accel, &priv->raw_accel);
```

Comment:
This should be `s->samples[i]`, not `s->samples[1]`, like it was before I broke it in f8420910a6ad.

Reply:
That is fixed in a later commit, I tried to make the refactor commit be completely a none functional change. I see now that I managed to change it from `0` to `1`.

Reply:
Ugh, I'm stupid... confused 1 and i. :-(

Reply:
Fixed!